site/content/extend/plugins/signing-plugins.md

```diff
@@ -0,0 +1,159 @@
+---
```

Comment:
I'm wondering if this documentation makes sense here. I don't see value in this being public (2/5) and it being alongside developer focused documentation (4/5). Maybe move it into https://github.com/mattermost/internal-docs or https://docs.mattermost.com/guides/core.html?

Reply:
This is still an outstanding concern for me

Reply:
@hanzei were you thinking we should wait before we finalize our certification process? This is intentionally put here to guide developers on how to sign their plugins. Some parts I agree can be moved/removed but a guide is necessary for them to know how to sign their own plugin, no?

Reply:
This documentation is needed if you
1. Run your own marketplace (We don't have amazing documentation for this and don't know anyone that wan't to do this right now)
2. Want to also sign your custom plugin for your own marketplace

I highly doubt that someone wants to do this in the near future. And if this becomes the case, we should have a more in debt documentation anyway.

Reply:
To take step back, we should answer these questions:
1. What is Plugin marketplace?
2. Why did we introduce plugin signatures? And how do they work?
3. What is the process of adding a plugin in the Mattermost Plugin Marketplace? And how can a developer add their plugin to the marketplace?
4. How can customers deploy their custom marketplace?

I think this concrete document should answer question 2. I'm 0/5 whether this document should be here or somewhere else.

Reply:
1. is answered by https://docs.mattermost.com/administration/plugins.html#plugin-marketplace
3. Is part of an ongoing discussion. I'm unsure where it should be documented.
4. should be answered at https://docs.mattermost.com/administration/plugins.html

With all of this in mind I would opt to move this to docs.mattermost.com. The audience is mostly admins and docs.mattermost.com are the admin docs.

Reply:
:+1: for adding this to https://docs.mattermost.com/administration/plugins.html#security

Reply:
Great. I'll add this to the existing docs once it's been merged.

Do we have any documentation answering "why and how" wrt the signatures?

Reply:
Well, this document kind of answers these two questions.

Reply:
Great - that's what I'd assumed but always best to make sure. :)

Reply:
Is this PR ready to be merged? No rush; but would like to make sure it's moved into the Admin docs as soon as possible. :)

Reply:
I think it is ready. @aaronrothschild thoughts?

Reply:
Why do you want to merge this PR if we want to have the documentation in the admin docs anyway? Can't we just resubmit this PR to the admin docs?

Reply:
Good question, @hanzei. My thinking is that it's neater to merge this PR and then take the content and move it into the admin guide once it's had all comments/issues/conversations resolved. I could be approaching it the wrong way, so happy to discuss further. :)